### ISSUE TITLE
Make HA discovery 'expire_after' field configurable per sensor

### ISSUE THREAD

**Issue by a-x-** [flyte/mqtt-io#163]
**Is your feature request related to a problem? Please describe.**
For instance dht22 is well-known buggy sensor. It frequently sends a unreadable trash.
It will be good to add an option to no say about read fails to MQTT.
Currently, it is unavailable most time

**Describe the solution you'd like**
When I parse it manually, I omit fails.
Also, I take 3 last values and avg it.

**Describe alternatives you've considered**
Do not use pi-mqtt-gpio

**Comment by a-x-** [comment]
![Uploading 18829629-6689-4D78-97F1-716BC8244FA7.png…]()

**Comment by a-x-** [comment]
![90F8B966-4DF0-45E5-9495-8F150541C709](https://user-images.githubusercontent.com/6201068/107576655-7eeb0d00-6c02-11eb-8126-37c2f00d3144.png)


Right after sending this message I again got `2021-02-11 00:45:16,226 mqtt_gpio (INFO): sensor_timer_thread: reading sensor 'wc_humidity' value 40.5` for a second, after a few minutes of nulls

**Comment by a-x-** [comment]
I don’t understand why it becomes Unavailable.
I cannot sniff anything weird over MQTT.
Also, code looks ok:
https://github.com/flyte/pi-mqtt-gpio/blob/35b8f2269d1217d239f6a1a7898a97efb50d5486/pi_mqtt_gpio/server.py#L752-L757

**Comment by a-x-** [comment]
I guess I found the reason.
pi-mqtt-gpio sends config:
```yaml
name: wc_humidity
unique_id: pi-mqtt-gpio-146b5dae_wc_climate_dht22_output_wc_humidity
state_topic: wc/sensor/wc_humidity
availability_topic: wc/status
payload_available: running
payload_not_available: dead
expire_after: 25
device:
  manufacturer: MQTT GPIO
  identifiers:
    - mqtt-gpio
    - pi-mqtt-gpio-146b5dae
  name: MQTT GPIO
platform: mqtt
```

where expire_after: 25 # sec
is too short

**Comment by a-x-** [comment]
Next, I found the code:
https://github.com/flyte/pi-mqtt-gpio/blob/35b8f2269d1217d239f6a1a7898a97efb50d5486/pi_mqtt_gpio/server.py#L1016

Looks I can hack it by increasing interval in my config. **UPD**: no I cannot, because I have to ask a sensor's value more frequently to dig anything useful.

**Comment by a-x-** [comment]
It might be helpful to add more straight config as I ask.
But it’s up to you

**Comment by flyte** [comment]
Ah, I see what you mean. The expire_after should be configurable in the config, yeah. I'll add it to the list, thanks.

**Comment by a-x-** [comment]
Thanks! I'll be waiting this config option

**Comment by flyte** [comment]
This is now implemented in the new version #178 so please give that a try.
